{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Great, now that we discussed a little let's continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given that the current approach utilized by the authors lacks reproducibility, we will explore an alternative method by leveraging nf-core pipelines for data analysis.\n",
    "\n",
    "Please explain, how we will achieve reproducibility for the course  with this approach.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will use a pipeline to streamline the process, using same packages versions etc., so that the process from input to output leaves little room for variance."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You have successfully downloaded 2 of the fastq files we will use in our study.\n",
    "\n",
    "What is the next step if we want to first have a count table and check the quality of our fastq files? What is the pipeline called to do so?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "nf-core/rnaseq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analyze the 2 files using an nf-core pipeline.\n",
    "\n",
    "What does this pipeline do?\n",
    "\n",
    "Which are the main tools that will be used in the pipeline?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "sheet = pd.read_csv('test_sheet/samplesheet/samplesheet.csv')\n",
    "sheet = sheet[['sample', 'fastq_1', 'fastq_2']]\n",
    "sheet['strandedness'] = ['auto', 'auto']\n",
    "sheet.to_csv('samplesheet_rna.csv', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mNextflow 24.04.4 is available - Please consider updating your version to it\u001b[m\n",
      "N E X T F L O W  ~  version 23.10.0\n",
      "Launching `https://github.com/nf-core/rnaseq` [compassionate_lamarr] DSL2 - revision: 4053b2ec17 [master]\n",
      "\u001b[33mWARN: Access to undefined parameter `monochromeLogs` -- Initialise it to a default value eg. `params.monochromeLogs = some_value`\u001b[39m\u001b[K\n",
      "\n",
      "\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "                                        \u001b[0;32m,--.\u001b[0;30m/\u001b[0;32m,-.\u001b[0m\n",
      "\u001b[0;34m        ___     __   __   __   ___     \u001b[0;32m/,-._.--~'\u001b[0m\n",
      "\u001b[0;34m  |\\ | |__  __ /  ` /  \\ |__) |__         \u001b[0;33m}  {\u001b[0m\n",
      "\u001b[0;34m  | \\| |       \\__, \\__/ |  \\ |___     \u001b[0;32m\\`-._,-`-,\u001b[0m\n",
      "                                        \u001b[0;32m`._,._,'\u001b[0m\n",
      "\u001b[0;35m  nf-core/rnaseq v3.15.1-g4053b2e\u001b[0m\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrevision                  : \u001b[0;32mmaster\u001b[0m\n",
      "  \u001b[0;34mrunName                   : \u001b[0;32mcompassionate_lamarr\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine           : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir                 : \u001b[0;32m/Users/jansteiger/Documents/Comp_workflows/day2\u001b[0m\n",
      "  \u001b[0;34mworkDir                   : \u001b[0;32m/Users/jansteiger/Documents/Comp_workflows/day2/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir                : \u001b[0;32m/Users/jansteiger/.nextflow/assets/nf-core/rnaseq\u001b[0m\n",
      "  \u001b[0;34muserName                  : \u001b[0;32mjansteiger\u001b[0m\n",
      "  \u001b[0;34mprofile                   : \u001b[0;32mtest,docker\u001b[0m\n",
      "  \u001b[0;34mconfigFiles               : \u001b[0;32m\u001b[0m\n",
      "\n",
      "\u001b[1mInput/output options\u001b[0m\n",
      "  \u001b[0;34minput                     : \u001b[0;32msamplesheet_rna.csv\u001b[0m\n",
      "  \u001b[0;34moutdir                    : \u001b[0;32mrna_seq\u001b[0m\n",
      "\n",
      "\u001b[1mReference genome options\u001b[0m\n",
      "  \u001b[0;34mgenome                    : \u001b[0;32mGRCm38\u001b[0m\n",
      "  \u001b[0;34mfasta                     : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/genome.fasta\u001b[0m\n",
      "  \u001b[0;34mgtf                       : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/genes_with_empty_tid.gtf.gz\u001b[0m\n",
      "  \u001b[0;34mgff                       : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/genes.gff.gz\u001b[0m\n",
      "  \u001b[0;34mgene_bed                  : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.bed\u001b[0m\n",
      "  \u001b[0;34mtranscript_fasta          : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/transcriptome.fasta\u001b[0m\n",
      "  \u001b[0;34madditional_fasta          : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/gfp.fa.gz\u001b[0m\n",
      "  \u001b[0;34mstar_index                : \u001b[0;32ms3://ngi-igenomes/igenomes//Mus_musculus/Ensembl/GRCm38/Sequence/STARIndex/\u001b[0m\n",
      "  \u001b[0;34mhisat2_index              : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/hisat2.tar.gz\u001b[0m\n",
      "  \u001b[0;34mrsem_index                : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/rsem.tar.gz\u001b[0m\n",
      "  \u001b[0;34msalmon_index              : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/salmon.tar.gz\u001b[0m\n",
      "\n",
      "\u001b[1mRead filtering options\u001b[0m\n",
      "  \u001b[0;34mbbsplit_fasta_list        : \u001b[0;32mhttps://raw.githubusercontent.com/nf-core/test-datasets/626c8fab639062eade4b10747e919341cbf9b41a/reference/bbsplit_fasta_list.txt\u001b[0m\n",
      "\n",
      "\u001b[1mUMI options\u001b[0m\n",
      "  \u001b[0;34mumitools_bc_pattern       : \u001b[0;32mNNNN\u001b[0m\n",
      "\n",
      "\u001b[1mAlignment options\u001b[0m\n",
      "  \u001b[0;34maligner                   : \u001b[0;32mhisat2\u001b[0m\n",
      "  \u001b[0;34mpseudo_aligner            : \u001b[0;32msalmon\u001b[0m\n",
      "  \u001b[0;34mmin_mapped_reads          : \u001b[0;32m5\u001b[0m\n",
      "\n",
      "\u001b[1mProcess skipping options\u001b[0m\n",
      "  \u001b[0;34mskip_bbsplit              : \u001b[0;32mfalse\u001b[0m\n",
      "\n",
      "\u001b[1mInstitutional config options\u001b[0m\n",
      "  \u001b[0;34mconfig_profile_name       : \u001b[0;32mTest profile\u001b[0m\n",
      "  \u001b[0;34mconfig_profile_description: \u001b[0;32mMinimal test dataset to check pipeline function\u001b[0m\n",
      "\n",
      "\u001b[1mMax job request options\u001b[0m\n",
      "  \u001b[0;34mmax_cpus                  : \u001b[0;32m8\u001b[0m\n",
      "  \u001b[0;34mmax_memory                : \u001b[0;32m16GB\u001b[0m\n",
      "  \u001b[0;34mmax_time                  : \u001b[0;32m6.h\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "If you use nf-core/rnaseq for your analysis please cite:\n",
      "\n",
      "* The pipeline\n",
      "  https://doi.org/10.5281/zenodo.1400710\n",
      "\n",
      "* The nf-core framework\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "  https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "\u001b[33mWARN: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
      "  Both '--gtf' and '--gff' parameters have been provided.\n",
      "  Using GTF file as priority.\n",
      "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
      "  '--transcript_fasta' parameter has been provided.\n",
      "  Make sure transcript names in this file match those in the GFF/GTF file.\n",
      "\n",
      "  Please see:\n",
      "  https://github.com/nf-core/rnaseq/issues/753\n",
      "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[39m\u001b[K\n",
      "\u001b[33mWARN: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
      "  When using '--additional_fasta <FASTA_FILE>' the aligner index will not\n",
      "  be re-built with the transgenes incorporated by default since you have \n",
      "  already provided an index via '--hisat2_index <INDEX>'.\n",
      "\n",
      "  Set '--additional_fasta <FASTA_FILE> --hisat2_index false --gene_bed false --save_reference'\n",
      "  to re-build the index with transgenes included and the index and gene BED file will be saved in\n",
      "  'results/genome/index/hisat2/' for re-use with '--hisat2_index'.\n",
      "\n",
      "  Ignore this warning if you know that the index already contains transgenes.\n",
      "\n",
      "  Please see:\n",
      "  https://github.com/nf-core/rnaseq/issues/556\n",
      "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[39m\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "\u001b[4A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "\u001b[5A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "\u001b[29A\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[54A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[54/361534] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[55A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[54/361534] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[55A\n",
      "executor >  local (1)\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[54/361534] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001b[K\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001b[K\n",
      "\u001b[55A\n",
      "-\u001b[0;35m[nf-core/rnaseq]\u001b[0;31m Pipeline completed with errors\u001b[0m-\n",
      "WARN: Killing running tasks (1)\n",
      "\n",
      "ERROR ~ Cannot invoke \"java.nio.file.Path.getFileSystem()\" because \"path\" is null\n",
      "\n",
      " -- Check script '/Users/jansteiger/.nextflow/assets/nf-core/rnaseq/./subworkflows/local/utils_nfcore_rnaseq_pipeline/../../nf-core/utils_nfcore_pipeline/main.nf' at line: 105 or see '.nextflow.log' file for more details\n"
     ]
    }
   ],
   "source": [
    "# post here the command you used to run nf-core/rnaseq\n",
    "!nextflow run nf-core/rnaseq -profile test,docker --input samplesheet_rna.csv --outdir rna_seq --genome GRCm38 --aligner hisat2 --max_cpus 8 --max_memory 16GB"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Performs quality control, trimming, alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As all other nf-core pipelines, the chosen pipeline takes in a samplesheet as input.\n",
    "\n",
    "Use Python and pandas to create the samplesheet for your 2 samples. Feel free to make use of the table you created earlier today.\n",
    "\n",
    "Choose your sample names wisely, they must be the connection of the results to the metadata. If you can't find the sample in the metadata later, the analysis was useless."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# post here the command you used to run nf-core/rnaseq\n",
    "!nextflow run nf-core/rnaseq -profile test,docker --input samplesheet_rna.csv --outdir rna_seq --genome GRCm38 --aligner hisat2 --max_cpus 8 --max_memory 16GB"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain all the parameters you set and why you set them in this way.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-profile:  \n",
    "--input: The samplesheet containing information to the fastq files  \n",
    "--outdir: Save results to this folder  \n",
    "--genome: The reference genome for mapping  \n",
    "--aligner: Specify usage of aligner, default being salmon  \n",
    "--max_cpus/memory: Specify CPU/memory usage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browsing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### How did the pipeline perform?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Good, took some time"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain the quality control steps. Are you happy with the quality and why. If not, why not.\n",
    "Please give additional information on : \n",
    "- ribosomal rRNA\n",
    "- Duplication\n",
    "- GC content\n",
    "\n",
    "What are the possible steps that could lead to poorer results?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Quality control:\n",
    "1. Strandedness prediction\n",
    "2. Check amount of exonic reads\n",
    "3. look at splice junctions\n",
    "4. calculate inner distance for two paired end reads\n",
    "5. Detect splice site saturation\n",
    "6. Assess read duplications\n",
    "7. Evaluate RNA integrity\n",
    "8. Assess mapping and coverage\n",
    "9. Check duplication rate vs expression\n",
    "10. Estimate number of redundant reads\n",
    "11. Step 2 again"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generally good quality reads\n",
    "\n",
    "rRNA very low, with some exceptions at around 1% and outliers are SNI_Sal 2 and SNI_Sal 4 (more than 20%, high contamination).\n",
    "\n",
    "Duplication generally fine, SNI_Sal 2/4 might be too high\n",
    "\n",
    "GC content around 50%\n",
    "\n",
    "Problems: technical problems, contamination"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Would you exclude any samples? If yes, which and why?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sham-oxy 1 maps has significantly less reads (thus also lower coverage), maybe exclude\n",
    "\n",
    "SNI-Sal 2 and 4 have high contamination, maybe combined with excessive duplication, also seem to skew PCA, maybe exclude"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### What would you now do to continue the experiment? What are the scientists trying to figure out? Which packages on R or python would you use?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I would look at the differentially expressed genes between to subgroups (now we have only two samples but generally...). This can be accomplished e.g. through the abundence analysis pipeline. The expression levels can be visualized (with SD within the group, use seaborn or matplotlib) and compared to the expression levels of other groups. With this analysis, we can see how the different conditions and phenotypes affect expression levels."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "work",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
